### PR TITLE
unique function to retrieve ID from satellite

### PR DESCRIPTION
**Pull request by Rocco83**
Fix the issue https://github.com/sideangleside/sat6-bootstrap/issues/41 .

Please doublecheck and let me know, seems working for me, but i was unable to test all conditions.

### CODE REVIEW

File: bootstrap.py [hunk 1]
```diff
@@ -300,61 +302,24 @@ def delete_json(url):
...
         sys.exit(2)
 
-
-def return_matching_domain_id(domain_name):
-    # Given a domain, find its id
-    myurl = "https://" + options.sat6_fqdn + ":" + API_PORT + "/api/v2/domains/?" + urlencode([('search', 'name=%s' % domain_name)])
+# Search in API 
+# given a search key, return the ID
+# api_name is the key in url for API name, search_key must contain also the key for search (name=, title=, ...)
+def return_matching_id(api_name, search_key, null_result_ok):
```
Comment (evgeni) [inline bootstrap.py]: maybe make `null_result_ok=False` the default?

File: bootstrap.py [hunk 2]
```diff
@@ -368,7 +333,6 @@ def return_puppetenv_for_hg(hg_id):
     else:
         return 'production'
 
-
```
Comment (evgeni) [inline bootstrap.py:371]: pep8 requires two empty lines between functions

**Comment by evgeni** [comment]
I think you missed a few occasions:
- https://github.com/Rocco83/sat6-bootstrap/blob/master/bootstrap.py#L336
- https://github.com/Rocco83/sat6-bootstrap/blob/master/bootstrap.py#L370
- https://github.com/Rocco83/sat6-bootstrap/blob/master/bootstrap.py#L438

otherwise LGTM :+1:

**Comment by Rocco83** [comment]
Thanks @evgeni for pointing out.
New code, not yet tested, i will do it ASAP... if someone else is willing to, very welcome :)

**Comment by Rocco83** [comment]
Forgot to add null_result_ok=False previously. the previous comment apply anyway.

**Comment by Rocco83** [comment]
I made a couple of test, seems working without problem.

**Pull request merged**